Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -626,9 +616,19 @@ bool CachedResourceLoader::canRequestAfterRedirection(CachedResource::Type type,
         return false;
     }
 
+    if (document() && MixedContentChecker::isMixedContent(document()->securityOrigin(), url)) {
+        CACHEDRESOURCELOADER_RELEASE_LOG("canRequestAfterRedirection: URL was not allowed because redirected content is insecure");
+        return false;
+    }
+
+    if (document() && !MixedContentChecker::isMixedContent(document()->securityOrigin(), preRedirectURL) && MixedContentChecker::isMixedContent(document()->securityOrigin(), url)) {
```

Comment:
I do not understand why we have these two if statements.
First, it seems they are somehow redundant (if the first one is false, the second is false as well).
Also, we already tried to upgrade the request in CachedResourceLoader::updateRequestAfterRedirection, so the check to checkInsecureContent below should be sufficient.